src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -357,64 +420,26 @@ export function getQuickFixesForCommand(
 	return fixes.length > 0 ? { fixes, onDidRunQuickFix, expectedCommands } : undefined;
 }
 
-export function convertToQuickFixOptions(quickFix: IExtensionTerminalQuickFix): ITerminalQuickFixOptions {
-	const type = quickFix.commandToRun ? 'command' : quickFix.linkToOpen ? 'opener' : undefined;
-	const options = {
-		id: quickFix.id,
-		commandLineMatcher: quickFix.commandLineMatcher,
-		outputMatcher: quickFix.outputMatcher,
-		type,
-		getQuickFixes: type === 'command' ? (matchResult: TerminalQuickFixMatchResult) => {
-			const matches = matchResult.outputMatch;
-			const commandToRun = quickFix.commandToRun;
-			if (!matches || !commandToRun) {
-				return;
-			}
-			const groups = matches.groups;
-			if (!groups) {
-				return;
-			}
-			const actions: TerminalQuickFixAction[] = [];
-			let fixedCommand = commandToRun;
-			for (const [key, value] of Object.entries(groups)) {
-				const varToResolve = '${group:' + `${key}` + '}';
-				if (!commandToRun.includes(varToResolve)) {
-					return [];
-				}
-				fixedCommand = fixedCommand.replaceAll(varToResolve, value);
-			}
-			if (fixedCommand) {
-				actions.push({
-					type: 'command',
-					id: quickFix.id,
-					command: fixedCommand,
-					addNewLine: true
-				});
-				return actions;
-			}
-			return;
-		} : (matchResult: TerminalQuickFixMatchResult) => {
-			const matches = matchResult.outputMatch;
-			const linkToOpen = quickFix.linkToOpen;
-			if (!matches || !linkToOpen) {
-				return;
-			}
-			const groups = matches.groups;
-			if (!groups) {
-				return;
-			}
-			let link = linkToOpen;
-			for (const [key, value] of Object.entries(groups)) {
-				const varToResolve = '${group:' + `${key}` + '}';
-				if (!linkToOpen?.includes(varToResolve)) {
-					return [];
-				}
-				link = link.replaceAll(varToResolve, value);
-			}
-			return link ? { type: 'opener', uri: URI.parse(link), id: quickFix.id } as ITerminalQuickFixOpenerAction : [];
-		},
-		exitStatus: quickFix.exitStatus,
-		source: quickFix.extensionIdentifier
+function convertToQuickFixOptions(selectorProvider: ITerminalQuickFixProviderSelector): IResolvedExtensionOptions {
+	return {
+		id: selectorProvider.selector.id,
+		type: 'resolved',
+		commandLineMatcher: selectorProvider.selector.commandLineMatcher,
+		outputMatcher: selectorProvider.selector.outputMatcher,
+		exitStatus: selectorProvider.selector.exitStatus,
+		getQuickFixes: selectorProvider.provider.provideTerminalQuickFixes
 	};
-	return options;
+}
+
+export function resolveAliases(commandLine: string, aliases?: string[][]): string {
+	if (!aliases) {
+		return commandLine;
+	}
+	for (const alias of aliases) {
+		if (alias.length !== 2) {
+			throw new Error('Not an alias');
+		}
```

Comment:
We'll want to silently fail I think, good general rule for any messages we get from the shell integration script/shell